main.js

```diff
@@ -0,0 +1,141 @@
+let encryptedWord = [];
+let player1 = document.getElementById("word-input");
+let player2 = document.getElementById("word-guessing");
+let wordToGuess;
+let mismatchedLetters = 0;
+let completeBoard = encrypted;
+let alphabetButtons = letters;
+let gallowsImages = images;
+let resetButton = restart;
+
+// hides the page of the player one and displays the elements needed to start the game
+function startGame(player1) {
+    wordToGuess = player1.input.value;
+    switchDisplay();
+    encryptLetters(wordToGuess);
+    statusWord();
+    displayAlphabet();
+    displayImages(mismatchedLetters);
+}
+
+// this is the main function of the game
+// for each clicked button:
+// → the encrypted word will be completed if it is the case
+// → hides the clicked button
+// → checks the status of the game
+// → displays the proper image
+function searchMatches(id) {
+    completeBoard.innerHTML = "";
+    let matchingLetters = completeWord(id);
+    markButton(id);
+    statusWord();
+    countLifes(matchingLetters);
+    displayImages(mismatchedLetters);
+}
+
+// for each letter in the word, a "_" will be displayed
+function encryptLetters(wordToGuess) {
+    for (let i = 0; i < wordToGuess.length; ++i) {
+        encryptedWord.push(" _ ");
+    }
+}
+
+// if all the letters in the word are matched, the game will be over
+// if all the lifes are wasted, the game is also over
+function countLifes(letterMatch) {
+    if (letterMatch == 0) {
+        ++mismatchedLetters;
+        if (mismatchedLetters == 8) {
+            mismatchedLetters = 9;
+            hangmanStatus(mismatchedLetters);
+        }
+    }
+    allLettersMatch(encryptedWord);
+}
+
+// displayes the letters of the alphabet as buttons
+// adds functionality to the buttons
+function displayAlphabet() {
+    for (let i = 0; i < alphabet.length; ++i) {
+        const currentButton = document.createElement("button");
+        alphabetButtons.appendChild(currentButton);
+        currentButton.innerHTML = alphabet[i];
+        currentButton.id = alphabet[i];
+        currentButton.addEventListener("click", ()=> searchMatches(currentButton.id));
+        currentButton.className = "btn btn-light";
+    }
+}
+
+// for each letter in the word, one "-" will be displayed
+function statusWord() {
+    for (let i = 0; i < encryptedWord.length; ++i) {
+        completeBoard.innerText += encryptedWord[i];
+    }   
+}
+
+// for each wrong letter, the man is closer to his death
+function displayImages(mismatchedLetters) {
+    gallowsImages.innerHTML = "";
+    let img = document.createElement("img");
+    img.src = gallowImages[mismatchedLetters];
+    gallowsImages.appendChild(img);
```

Comment:
Can't we only change the `src` style property instead of creating a new HTML element every time?

Reply:
right